src/test/java/it/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMIT.java

```diff
@@ -70,6 +70,17 @@ public void testCheckout() throws Exception {
         assertTrue(build.getWorkspace().child("add_file").isDirectory());
     }
 
+    @Test
+    public void testCheckoutWithSsh() throws Exception {
+        project.setScm(createSCMWithSshCredentials());
+        project.save();
+
+        FreeStyleBuild build = project.scheduleBuild2(0).get();
+
+        assertEquals(SUCCESS, build.getResult());
```

Comment:
I think this is too optimistic, I don't think you can immediately assert the success of the build here, I believe we've got infrastructure for this sort of stuff to allow you to wait and appropriate amount of time before failing if the build is not successful.

Reply:
Do we have tests for pipelines and multibranch pipelines as well? If we already have test for that I think it would be good to include ssh based cloning in those tests (as in create new tests). If we don't have tests for that, well, that's a glaring hole we need to plug :D

Reply:
> I believe we've got infrastructure for this sort of stuff to allow you to wait and appropriate amount of time before failing if the build is not successful.
I might be mistake, but from my understanding, that's what the `.get()` call does- it waits before there is a build result (SUCCESS or FAILURE), before reporting.

Reply:
There is no wait
```Java
/**
@return The status of the build, if it has completed or some build step has set a status; may be null if the build is ongoing. **/

 @Exported
    public @CheckForNull Result getResult() {
        return result;
    }
```
I believe we have some stuff already for this type of scenario

Reply:
Oh, derp, you meant on the project!